pkg/apis/pingcap/v1alpha1/tidbmonitor.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+func (tm *TidbMonitor) PrometheusPortName() *string {
+	return tm.Spec.Prometheus.Service.PortName
+}
+
+func (tm *TidbMonitor) GrafanaPortName() *string {
+	if tm.Spec.Grafana != nil {
+		return tm.Spec.Grafana.Service.PortName
+	}
+	return nil
+}
+
+func (tm *TidbMonitor) ReloaderPortName() *string {
+	return tm.Spec.Reloader.Service.PortName
+}
```

Comment:
move all default values here, and in controller logic we can simply use `TidbMonitor. (PrometheusPortName| GrafanaPortName| ReloaderPortName)`? like https://github.com/pingcap/tidb-operator/blob/d314b251735693a000088d7011f9a5b2932f579f/pkg/manager/member/pump_member_manager.go#L373-L379